pkg/cmd/api/api.go

```diff
@@ -426,23 +426,50 @@ func parseErrorResponse(r io.Reader, statusCode int) (io.Reader, string, error)
 
 	var parsedBody struct {
 		Message string
-		Errors  []struct {
-			Message string
-		}
+		Errors  []json.RawMessage
 	}
 	err = json.Unmarshal(b, &parsedBody)
 	if err != nil {
 		return r, "", err
 	}
-
 	if parsedBody.Message != "" {
 		return bodyCopy, fmt.Sprintf("%s (HTTP %d)", parsedBody.Message, statusCode), nil
-	} else if len(parsedBody.Errors) > 0 {
-		msgs := make([]string, len(parsedBody.Errors))
-		for i, e := range parsedBody.Errors {
+	}
+
+	type errorMessage struct {
+		Message string
+	}
+	var objectErrors []errorMessage
+	var stringErrors []string
+	for _, rawErr := range parsedBody.Errors {
+		if len(rawErr) == 0 {
+			continue
+		}
+		if rawErr[0] == '{' {
+			var objectError errorMessage
+			err := json.Unmarshal(rawErr, &objectError)
+			if err != nil {
+				return r, "", err
+			}
+			objectErrors = append(objectErrors, objectError)
```

Comment:
Tip: why keep a separate slice of `objectErrors` when you can just append `objectError.Message` to `stringErrors` and simplify the whole logic

Reply:
Less code and unified error handling like in my first approach 👍🏻